### PR TITLE
Support Azure Functions .NET isolated workers project type

### PR DESCRIPTION
**Pull request by maartenba**
This PR adds support for the latest Azure Functions isolated workers project type.

It's part of #413, and provides:
* Gutter marks
* Run configuration/run support
* Completion for timer triggers

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-intellij/rider/ReSharper.Azure/src/Azure.Daemon/FunctionApp/FunctionAppDaemonHost.cs [hunk 1]
```diff
@@ -1,17 +1,17 @@
 // Copyright (c) 2020 JetBrains s.r.o.
-// <p/>
```
Comment (sdubov) [inline PluginsAndFeatures/azure-toolkit-for-intellij/rider/ReSharper.Azure/src/Azure.Daemon/FunctionApp/FunctionAppDaemonHost.cs:2]: Do you think we need to use shorten headers, like in the upstream?

File: PluginsAndFeatures/azure-toolkit-for-intellij/rider/ReSharper.Azure/src/Azure.Project/FunctionApp/FunctionAppProjectDetector.cs [hunk 2]
```diff
@@ -34,8 +34,25 @@ namespace JetBrains.ReSharper.Azure.Project.FunctionApp
 {
     public static class FunctionAppProjectDetector
     {
-        private static readonly NugetId ExpectedPackageForNet50 = new NugetId("Microsoft.Azure.Functions.Worker");
-        private static readonly NugetId ExpectedPackageForOlder = new NugetId("Microsoft.NET.Sdk.Functions");
+        static class DefaultWorker
+        {
+            public static readonly NugetId ExpectedPackage = new NugetId("Microsoft.NET.Sdk.Functions");
```
Comment (sdubov) [inline PluginsAndFeatures/azure-toolkit-for-intellij/rider/ReSharper.Azure/src/Azure.Project/FunctionApp/FunctionAppProjectDetector.cs]: Could you please provide a more descriptive variable name, e.g. `FunctionNuGetPackageId`?

File: PluginsAndFeatures/azure-toolkit-for-intellij/rider/ReSharper.Azure/src/Azure.Project/FunctionApp/FunctionAppProjectDetector.cs [hunk 3]
```diff
@@ -34,8 +34,25 @@ namespace JetBrains.ReSharper.Azure.Project.FunctionApp
...
+            
+            public static bool HasFunctionsPackageReference(IProject project, TargetFrameworkId targetFrameworkId)
+            {
+                return project.GetPackagesReference(DefaultWorker.ExpectedPackage, targetFrameworkId) != null;
+            }
+        }
+        
+        static class IsolatedWorker
+        {
+            public static readonly NugetId ExpectedPackage = new NugetId("Microsoft.Azure.Functions.Worker");
```
Comment (sdubov) [inline PluginsAndFeatures/azure-toolkit-for-intellij/rider/ReSharper.Azure/src/Azure.Project/FunctionApp/FunctionAppProjectDetector.cs]: Same comment here.

File: PluginsAndFeatures/azure-toolkit-for-intellij/rider/ReSharper.Azure/src/Azure.Project/FunctionApp/FunctionAppProjectDetector.cs [hunk 4]
```diff
@@ -110,7 +129,8 @@ public static bool IsAzureFunctionsProject([NotNull] IProject project)
                 .FirstNotNull());
 
             // 2) Check expected package reference.
-            var hasExpectedPackageReference = HasFunctionsPackageReference(project, targetFrameworkId);
+            var hasExpectedPackageReference = DefaultWorker.HasFunctionsPackageReference(project, targetFrameworkId) ||
```
Comment (sdubov) [inline PluginsAndFeatures/azure-toolkit-for-intellij/rider/ReSharper.Azure/src/Azure.Project/FunctionApp/FunctionAppProjectDetector.cs]: Could you please add some comment here with a reference to the function upgrade link to understand the difference between Isolated worker and default worker?

File: PluginsAndFeatures/azure-toolkit-for-intellij/rider/test-integration/org/cases/markup/functionapp/csharp/isolatedWorker/AzureFunctionAppSyntaxHighlightingTest.kt [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+import com.intellij.lang.annotation.HighlightSeverity
+import com.jetbrains.rdclient.testFramework.waitForDaemon
+import com.jetbrains.rider.test.annotations.TestEnvironment
+import com.jetbrains.rider.test.base.BaseTestWithMarkup
+import com.jetbrains.rider.test.enums.CoreVersion
+import org.testng.annotations.Test
+
+@TestEnvironment(coreVersion = CoreVersion.DOT_NET_5)
+class AzureFunctionAppSyntaxHighlightingTest : BaseTestWithMarkup() {
```
Comment (sdubov) [inline PluginsAndFeatures/azure-toolkit-for-intellij/rider/test-integration/org/cases/markup/functionapp/csharp/isolatedWorker/AzureFunctionAppSyntaxHighlightingTest.kt]: We spawn a lot of duplicates in tests in this case. Instead, I would suggest to move all tests into some base class and create two inherited classes for 5.0 and 3.1. You can take a look on example in `RiderTemplatesTestCore` in Rider project.

File: PluginsAndFeatures/azure-toolkit-for-intellij/.idea/vcs.xml [hunk 6]
```diff
@@ -1,5 +1,13 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <project version="4">
+  <component name="GithubSharedProjectSettings">
```
Comment (sdubov) [inline PluginsAndFeatures/azure-toolkit-for-intellij/.idea/vcs.xml]: Do we need to commit this?

**Review by sdubov — CHANGES_REQUESTED** [review]

File: PluginsAndFeatures/azure-toolkit-for-intellij/rider/ReSharper.Azure/src/Azure.Daemon/FunctionApp/FunctionAppDaemonHost.cs (hunk 1 above)
Comment (maartenba) [reply PluginsAndFeatures/azure-toolkit-for-intellij/rider/ReSharper.Azure/src/Azure.Daemon/FunctionApp/FunctionAppDaemonHost.cs:2]: Not sure, could work. Perhaps we can do an update at some point in a separate PR when things aren't changing that much?

**Review by sdubov — APPROVED** [review]

File: PluginsAndFeatures/azure-toolkit-for-intellij/.idea/vcs.xml (hunk 6 above)
Comment (maartenba) [reply PluginsAndFeatures/azure-toolkit-for-intellij/.idea/vcs.xml]: Not sure, IntelliJ keeps adding it. I'll revert the change.

File: PluginsAndFeatures/azure-toolkit-for-intellij/rider/test-integration/org/cases/markup/functionapp/csharp/isolatedWorker/AzureFunctionAppSyntaxHighlightingTest.kt (hunk 5 above)
Comment (maartenba) [reply PluginsAndFeatures/azure-toolkit-for-intellij/rider/test-integration/org/cases/markup/functionapp/csharp/isolatedWorker/AzureFunctionAppSyntaxHighlightingTest.kt]: Reworked all tests to use this approach (where applicable).

**Pull request merged**